src/store/flow/block-types/MobilePrimitives_NumericResponseBlockStore.ts

```diff
@@ -68,21 +69,45 @@ export const actions: ActionTree<IFlowsState, IRootState> = {
         prompt: blankResource.uuid,
         validation_minimum: undefined,
         validation_maximum: undefined,
+        ...await dispatch('initiateExtraVendorConfig'),
       },
       tags: [],
       vendor_metadata: {},
     })
   },
 
+  /**
+   * Override this in the consumer side to add extra config props to avoid the validation saying we have missing prop at the creation
+   * eg: {
+   *   prop1: undefined,
+   *   prop2: undefined,
+   * }
+   */
+  async initiateExtraVendorConfig(_ctx: unknown): Promise<object> {
+    return {}
+  },
+
   handleBranchingTypeChangedToUnified({dispatch}, {block}: {block: IBlock}) {
     dispatch('flow/block_convertExitFormationToUnified', {
       blockId: block.uuid,
       test: 'is_number(block.value)',
     }, {root: true})
   },
 
-  validate({rootGetters}, {block, schemaVersion}: {block: IBlock, schemaVersion: string}) {
-    const validationResults = validateCommunityBlock({block, schemaVersion})
+  /**
+   * Validate the vendor block (Consumer block)
+   * By overriding this action in the consumer side, we will be able to customize it using different json schema for eg.
+   *
+   * Important: This will be overridden in the consumer side, so DO NOT add generic validations here,
+   * instead edit the `validate()` if needed.
+   */
+  async validateVendorBlock({rootGetters}, {block, schemaVersion}: {block: IBlock, schemaVersion: string}): Promise<IValidationStatus> {
+    const blockTypeWithoutNameSpace = block.type.split('.')[block.type.split('.').length - 1]
+    return validateCommunityBlock({block, schemaVersion, customBlockJsonSchema: require(`@floip/flow-runner/dist/resources/validationSchema/${schemaVersion}/I${blockTypeWithoutNameSpace}Block.json`)})
```

Comment:
I think that validateCommunityBlock and validateVendorBlock are maybe no longer the right names. Here is a suggestion:

- `validateVendorBlock` -> `fetchBlockSchemaAndValidate`
- `validateCommunityBlock` becomes something like `validateBlockWithSchema`

Then I think you still only need to override the method now called `fetchBlockSchemaAndValidate` with vendor specific behaviour.

Reply:
Agree here, thanks for noticing. I renamed them & updated the doc too, 🙏